### PR TITLE
Parallelise memcache puts.

### PR DESCRIPTION
**Pull request by tomwilkie**

### CODE REVIEW

**Comment by juliusv** [comment]
:+1:

File: frankenstein/memcache_client.go [hunk 1]
```diff
@@ -205,3 +205,20 @@ func (c *MemcacheClient) StoreChunkData(userID string, chunk *wire.Chunk) error
...
+	for _, chunk := range chunks {
+		go func(chunk *wire.Chunk) {
+			errs <- c.StoreChunkData(userID, chunk)
+		}(&chunk)
+	}
+	var errOut error
+	for i := 0; i < len(chunks); i++ {
+		if err := <-errs; err != nil {
+			errOut = err
+		}
```
Comment (jml) [inline frankenstein/memcache_client.go:221]: It kind of sucks that we're collecting all of the errors but discarding the information for most of them. Is it possible to aggregate them somehow before returning?

File: circle.yml [hunk 2]
```diff
@@ -23,5 +23,5 @@ deployment:
       - docker login -e $DOCKER_EMAIL -u $DOCKER_LOGIN -p $DOCKER_PASSWORD
       - docker push tomwilkie/prometheus:$(git rev-parse --abbrev-ref HEAD)-$(git rev-parse --short HEAD)
       - docker push tomwilkie/frankenstein:$(git rev-parse --abbrev-ref HEAD)-$(git rev-parse --short HEAD)
-      - wcloud deploy tomwilkie/prometheus:$(git rev-parse --abbrev-ref HEAD)-$(git rev-parse --short HEAD)
-      - wcloud deploy tomwilkie/frankenstein:$(git rev-parse --abbrev-ref HEAD)-$(git rev-parse --short HEAD)
+      - wcloud deploy -u circle tomwilkie/prometheus:$(git rev-parse --abbrev-ref HEAD)-$(git rev-parse --short HEAD)
+      - wcloud deploy -u circle tomwilkie/frankenstein:$(git rev-parse --abbrev-ref HEAD)-$(git rev-parse --short HEAD)
```
Comment (jml) [inline circle.yml:27]: Unrelated change?

**Comment by jml** [comment]
Couple of comments but otherwise LGTM.

File: circle.yml (hunk 2 above)
Comment (tomwilkie) [inline circle.yml:27]: Yeah unrelated change but harmless...

File: frankenstein/memcache_client.go (hunk 1 above)
Comment (tomwilkie) [inline frankenstein/memcache_client.go:221]: Could do, but I've not found it to be particularly valuable in the past.  I only collect them all for correctness.

**Comment by jml** [comment]
LGTM.
